### ISSUE TITLE
Having typescript issue for the toaster component

### ISSUE THREAD

**Issue by jeevasusej** [timolins/react-hot-toast#248]
I have tried to use the following component but I am having typescript error. 

```
<Toaster
  position="top-center"
  reverseOrder={false}
  gutter={8}
  containerClassName=""
  containerStyle={{}}
  toastOptions={{
    // Define default options
    className: '',
    duration: 5000,
    style: {
      background: '#363636',
      color: '#fff',
    },

    // Default options for specific types
    success: {
      duration: 3000,
      theme: {
        primary: 'green',
        secondary: 'black',
      },
    },
  }}
/>
```

![image](https://user-images.githubusercontent.com/19813131/206675406-9c4b4c9b-5d40-4dc4-b2f1-df1ce1e9a60d.png)

**Comment by guotie** [comment]
I have a similar question.

When I use the example `useToaster`:
```
import { useToaster } from 'react-hot-toast/headless'

export const Notifications = () => {
  const { toasts, handlers } = useToaster()
  const { startPause, endPause, calculateOffset, updateHeight } = handlers

  return (
    <div
      style={{
        position: 'fixed',
        bottom: 8,
        right: 8,
      }}
      onMouseEnter={startPause}
      onMouseLeave={endPause}
    >
      {toasts.map((toast) => {
        const offset = calculateOffset(toast, {
          reverseOrder: false,
          gutter: 8,
        })

        const ref = (el) => {
          if (el && typeof toast.height !== 'number') {
            const height = el.getBoundingClientRect().height
            updateHeight(toast.id, height)
          }
        }
        return (
          <div
            key={toast.id}
            ref={ref}
            style={{
              position: 'absolute',
              width: '200px',
              background: 'papayawhip',
              transition: 'all 0.5s ease-out',
              opacity: toast.visible ? 1 : 0,
              transform: `translateY(${offset}px)`,
            }}
            {...toast.ariaProps}
          >
            {toast.message}
          </div>
        )
      })}
    </div>
  )
}

```

and typescript error:
```
Type 'ValueOrFunction<Renderable, Toast>' is not assignable to type 'ReactNode'.

```

**Comment by BoKleynen** [comment]
I'm seeing the same problem since upgrading from react 17 to 18

**Comment by timolins** [comment]
You need to use `iconTheme` instead of `theme`:
```
   success: {
      duration: 3000,
      iconTheme: {
        primary: 'green',
        secondary: 'black',
      },
    },
```

**Comment by timolins** [comment]
@guotie @BoKleynen Feel free to open a new issue for this, as it's a different one.

TLDR:

You need to wrap `toast.message` with `resolveValue`. 

```jsx
import { resolveValue } from "react-hot-toast/headless";

<div>{resolveValue(toast.message, toast)}</div>

```
